src/main/scala/com/databricks/spark/avro/LogicalTypeConverters.scala

```diff
@@ -15,17 +15,48 @@
  */
 package com.databricks.spark.avro
 
+import java.math.BigDecimal
+import java.nio.ByteBuffer
+import org.apache.avro.{Schema, _}
+import org.apache.avro.LogicalTypes.Decimal
+import org.apache.spark.sql.types._
+
 /**
   * This object contains methods that are used to convert
   * Avro Logical types to Spark SQL schemas and vice versa.
   */
+
 private object LogicalTypeConverters {
 
-  private[avro] def convertFieldTypeToLogical = Unit
+  private[avro] def convertFieldTypeToLogical(): Unit = {
+
+  }
 
-  private[avro] def convertToLogicalValue = Unit
+  private[avro] def convertToLogicalValue(item: Any): ByteBuffer = {
+    val bigItem = item.asInstanceOf[java.math.BigDecimal]
+    ByteBuffer.wrap(bigItem.unscaledValue().toByteArray)
+  }
 
-  private[avro] def toSqlType = Unit
+  private[avro] def toSqlType(datatype: LogicalType): Option[DecimalType] = datatype match {
+    case decimal: Decimal => Some(DecimalType(decimal.getPrecision, decimal.getScale))
+    case _ => None
+  }
 
-  private[avro] def toSql = Unit
-}
+  private[avro] def toSql(logicalType: LogicalType, item: Any, schema: Schema): Option[BigDecimal] = logicalType match {
+    case decimalType: Decimal => Some(new Conversions.DecimalConversion().fromBytes(
+      item.asInstanceOf[ByteBuffer],
+      schema,
+      schema.getProp("scale") match {
```

Comment:
Probably best to move the `decimalFromBytes` from the SchemaConverters and use it here as the `fromBytes` Avro lib doesn't work as expected